File: engine/scripts/gen_sop_tex.py
#!/usr/bin/env python

import sop, sys
from string import Template

the_sop = sop.get_sop_meta(sys.argv[1])

front = Template("""\\input{preamble.tex}

\\author{$author}
\\title{$title}

\\begin{document}

\\renewcommand{\\bibname}{References}

%\\frontmatter
\\mainmatter

\\maketitle

\\begin{abstract}
    To be added...
\\end{abstract}
\\clearpage

\\tableofcontents*
\\clearpage

\\chapter{Introduction}

To be added...""").safe_substitute(author = the_sop["author"],
                                 title = the_sop["title"])

back = """\n
\\backmatter

\\end{document}
"""


print front

for part in the_sop["chapters"]:
    partname, chapters = part.popitem()
    print "\n\n\\part{" + partname + "}\n"
    for ch in chapters:
        print "\\include{" + ch + "-frag}"

print back

File: engine/scripts/gen_sop_tex.R
#!/usr/bin/env Rscript

the_sop <- commandArgs(trailingOnly = TRUE)[1]
sop_meta <- yaml::yaml.load_file(the_sop)

front <- "\\input{preamble.tex}\n
\\title{%s}
\\author{%s}\n
\\begin{document}

\\renewcommand{\\bibname}{References}

%%\\frontmatter
\\mainmatter
\\maketitle

\\begin{abstract}
To be added...
\\end{abstract}
\\clearpage

\\tableofcontents*
\\clearpage

\\chapter{Introduction}
To be added...\n"

front <- sprintf(front, sop_meta$title, sop_meta$author)

back <- "\n\\backmatter
\\end{document}"

chapters <- sapply(sop_meta$chapters, function(part) {
  paste0("\n\\part{", names(part), "}\n",
         paste0(paste0("\\include{", unlist(part), "-frag}"), collapse = "\n"),
         "\n"
  )
})

cat(front, chapters, back, "\n", sep = "")
